### ISSUE TITLE
RuntimeError: The expanded size of the tensor (3) must match the existing size (8) at non-singleton dimension 1.

### ISSUE THREAD

**Issue by LieDeath** [pkuliyi2015/multidiffusion-upscaler-for-automatic1111#168]
It is a replicated issue to #154, #155, #156. **However, it is never fixed on my device.** Even if I reinstall completely new a1111's WEBUI and multidiffusion-upscaler-for-automatic1111 extension. The problem still exist in the latest 6931b89.
**The problem exist when hires.fix is on, with upscaler R-ESRGAN 4x+ Anime6B.** When the program run to encode process, it just crashed.
My runtime configs is torch2.0.0-cu118 and xformers0.0.17rc482, the hardware info is RTX3070Laptop with Studio Driver 531.61. No other extensions is used.
The traceback are these below.
```
Traceback (most recent call last):
  File "D:\AI7\stable-diffusion-webui\modules\call_queue.py", line 57, in f
    res = list(func(*args, **kwargs))
  File "D:\AI7\stable-diffusion-webui\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "D:\AI7\stable-diffusion-webui\modules\txt2img.py", line 56, in txt2img
    processed = process_images(p)
  File "D:\AI7\stable-diffusion-webui\modules\processing.py", line 515, in process_images
    res = process_images_inner(p)
  File "D:\AI7\stable-diffusion-webui\modules\processing.py", line 669, in process_images_inner
    samples_ddim = p.sample(conditioning=c, unconditional_conditioning=uc, seeds=seeds, subseeds=subseeds, subseed_strength=p.subseed_strength, prompts=prompts)
  File "D:\AI7\stable-diffusion-webui\modules\processing.py", line 942, in sample
    samples = self.sd_model.get_first_stage_encoding(self.sd_model.encode_first_stage(decoded_samples))
  File "D:\AI7\stable-diffusion-webui\modules\sd_hijack_utils.py", line 17, in <lambda>
    setattr(resolved_obj, func_path[-1], lambda *args, **kwargs: self(*args, **kwargs))
  File "D:\AI7\stable-diffusion-webui\modules\sd_hijack_utils.py", line 28, in __call__
    return self.__orig_func(*args, **kwargs)
  File "D:\AI7\stable-diffusion-webui\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\AI7\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 830, in encode_first_stage
    return self.first_stage_model.encode(x)
  File "D:\AI7\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\autoencoder.py", line 83, in encode
    h = self.encoder(x)
  File "D:\AI7\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\AI7\stable-diffusion-webui\extensions\multidiffusion-upscaler-for-automatic1111\scripts\vae_optimize.py", line 377, in __call__
    return self.vae_tile_forward(x)
  File "D:\AI7\stable-diffusion-webui\extensions\multidiffusion-upscaler-for-automatic1111\scripts\vae_optimize.py", line 264, in wrapper
    ret = fn(*args, **kwargs)
  File "D:\AI7\stable-diffusion-webui\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\AI7\stable-diffusion-webui\extensions\multidiffusion-upscaler-for-automatic1111\scripts\vae_optimize.py", line 618, in vae_tile_forward
    result[:, :, out_bboxes[i][2]:out_bboxes[i][3], out_bboxes[i][0]:out_bboxes[i][1]] = crop_valid_region(tile, in_bboxes[i], out_bboxes[i], is_decoder)
RuntimeError: The expanded size of the tensor (3) must match the existing size (8) at non-singleton dimension 1.  Target sizes: [1, 3, 160, 160].  Tensor sizes: [8, 160, 160]

[Tiled VAE]: Executing Encoder Task Queue:  25%|█████████▎                           | 91/364 [00:00<00:02, 109.40it/s]
```

**Comment by LieDeath** [comment]
It even failed in i2i, annoying yet.

**Comment by nitorita** [comment]
I keep getting this error as well!!!!!!! And also a "Maximum Recusion Depth Exceeded" error too!!

Did somebody break something with the latest version? How do I revert back??

**Comment by LieDeath** [comment]
You can fall back to 5da750b, which is the latest workable commit.
run the code below in git bash at the repository folder.
```
git reset --hard 5da750b
```

**Comment by Kahsolt** [comment]
Could you please post a screenshot of your file `vae_optimize.py` around line 618?
A recent merged PR seems very problematic in compatibility, though everything still works on my machine, there are many broken reports. More error log and traceback information is highly welcome to help me debugging.

**Comment by LieDeath** [comment]
> Could you please post a screenshot of your file `vae_optimize.py` around line 618? A recent merged PR seems very problematic in compatibility, though everything still works on my machine, there are many broken reports. More error log and traceback information is highly welcome to help me debugging.

Here it is. Well, there is no more error log on my computer
![image](https://user-images.githubusercontent.com/115352572/236111634-c10d33bc-f8ef-447c-b62f-670df3ae031d.png)

**Comment by DimmSum88** [comment]
Here's mine
![pythonw_FMgiPeYL0U](https://user-images.githubusercontent.com/95658942/236112212-cc453c7d-0381-48ae-ba66-ae1378062502.png)

**Comment by DimmSum88** [comment]
> Could you please post a screenshot of your file `vae_optimize.py` around line 618? A recent merged PR seems very problematic in compatibility, though everything still works on my machine, there are many broken reports. More error log and traceback information is highly welcome to help me debugging.

Here's mine

![pythonw_FMgiPeYL0U](https://user-images.githubusercontent.com/95658942/236112571-9b1bc456-ee82-49fd-b8f5-4e289169b7d1.png)

**Comment by Kahsolt** [comment]
So sir @LieDeath, you are not on the lateset commit, see @DimmSum88 gives the right thing :)

The broken: 7253cb449c85e2d2317ba541bf770ef372a807a4
The fixed: 860f8a405193bcd992e21d82e43fa18137bc4923
The latest: fbe34db704736e5745c3a4e855939508eef344ce

**Comment by LieDeath** [comment]
> So sir @LieDeath, you are not on the lateset commit, see @DimmSum88 gives the right thing :)
> 
> The broken: [7253cb4](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/commit/7253cb449c85e2d2317ba541bf770ef372a807a4) The fixed: [860f8a4](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/commit/860f8a405193bcd992e21d82e43fa18137bc4923) The latest: [fbe34db](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/commit/fbe34db704736e5745c3a4e855939508eef344ce)

Thank you! The problem solved.
When I wrote it, the one on my computer was still the older one :(
